src/adaptertrimmer.cpp

```diff
@@ -55,80 +56,100 @@ int AdapterTrimmer::trimByMultiSequences(Read* r, FilterResult* fr, vector<strin
     return trimmed;
 }
 
-int AdapterTrimmer::searchAdapter(string* read, string& adapter, double edMax, int searchStart, int searchLen, bool asLeftAsPossible, bool asRightAsPossible) {
+int AdapterTrimmer::searchAdapter(string *read, const string &adapter, double edMax, int searchStart, int searchLen, bool asLeftAsPossible, bool asRightAsPossible)
+{
     int minMismatch = 99999; // initialized with a large mismatch
     int pos = -1;
     // for the best match
-    const char* adata = adapter.c_str();
-    const char* rdata = read->c_str();
+    const char *adata = adapter.c_str();
+    const char *rdata = read->c_str();
     int rlen = read->length();
     int alen = adapter.length();
 
     int threshold = round(edMax * alen);
 
     int searchEnd = rlen;
-    if(searchLen > 0) {
+    if (searchLen > 0)
+    {
         searchEnd = min(rlen, searchLen + searchStart);
     }
 
-    if(searchStart + alen > rlen )
+    if (searchStart + alen > rlen)
         return -1;
 
-    if(asLeftAsPossible) {
+    if (asLeftAsPossible)
+    {
         // go from left, and return immediatedly if find a mismatch < threshold
-        for(int p = searchStart; p < searchEnd - alen; p++) {
+        // if (false)
+        for (int p = searchStart; p < searchEnd - alen; p++)
+        {
             int mismatch = 0;
-            for(int i=0; i<alen; i++) {
-                if(rdata[p+i] != adata[i])
+            for (int i = 0; i < alen; i++)
+            {
+                if (rdata[p + i] != adata[i])
                     mismatch++;
             }
-            if(mismatch <= threshold) {
+            if (mismatch <= threshold)
+            {
                 return p;
             }
-            if(mismatch <= minMismatch ) {
+            if (mismatch <= minMismatch)
+            {
                 minMismatch = mismatch;
                 pos = p;
             }
         }
-    } else if(asRightAsPossible && searchEnd > alen) {
+    }
+    else if (asRightAsPossible && searchEnd > alen)
+    {
         // go from right, and return immediatedly if find a mismatch <= threshold
-        for(int p = searchEnd - alen - 1 ; p >= searchStart ; p--) {
+        for (int p = searchEnd - alen; p >= searchStart; p--)
```

Comment:
Note that this line has changed.
Previously it was checking it was checking the last nucleotide in the `rdata` variable.